modules/src/mock_context.rs

```diff
@@ -0,0 +1,199 @@
+
+/// Mock for a context. Used in testing handlers of all modules.
+#[derive(Clone, Debug)]
+pub struct MockContext {
+    /// Maximum size of the history.
+    max_history_size: usize,
+
+    /// Highest height of the headers in the history.
+    latest_height: Height,
+
+    /// A list of `max_history_size` headers, ascending order by their height (latest is last).
+    history: Vec<MockHeader>,
+
+    // All the connections in the store.
+    connections: HashMap<ConnectionId, ConnectionEnd>,
+
+    /// The set of all clients, indexed by their id.
+    clients: HashMap<ClientId, MockClientRecord>,
+
+    // Association between client ids and connection ids.
+    client_connections: HashMap<ClientId, ConnectionId>,
+}
+
+impl MockContext {
+    pub fn new(max_history_size: usize, latest_height: Height) -> Self {
+        // Compute the number of headers to store. If h is 0, nothing is stored.
+        let n = min(max_history_size as u64, latest_height.value());
+        MockContext {
+            max_history_size,
+            latest_height,
+            history: (0..n)
+                .rev()
+                .map(|i| MockHeader((latest_height.value() - i).try_into().unwrap()))
+                .collect(),
+            connections: Default::default(),
+            clients: Default::default(),
+            client_connections: Default::default(),
+        }
+    }
+
+    /// Internal interface for associating a connection to this context.
+    pub fn with_connection(
+        self,
+        connection_id: ConnectionId,
+        connection_end: ConnectionEnd,
+    ) -> Self {
+        let mut connections = self.connections.clone();
+        connections.insert(connection_id, connection_end);
+        Self {
+            connections,
+            ..self
+        }
+    }
+
+    /// Internal interface for associating a client record to this context.
+    /// Given a client id and a height, registers a new client in the context and also associates
+    /// to this client a mock client state and a mock consensus state for height `height`. The type
+    /// of this client is implicitly assumed to be Mock.
+    pub fn with_client(self, client_id: &ClientId, height: Height) -> Self {
+        let mut clients = self.clients.clone();
+
+        let mut client_record = MockClientRecord {
+            client_type: ClientType::Mock,
+            client_state: MockClientState(MockHeader(height)),
+            consensus_states: HashMap::with_capacity(1),
+        };
+        client_record
+            .consensus_states
+            .insert(height, MockConsensusState(MockHeader(height)));
+        clients.insert(client_id.clone(), client_record);
+
+        Self { clients, ..self }
+    }
+
+    /// Internal interface. Accessor for a header of the local (host) chain of this context.
+    /// May return `None` if the header for the requested height does not exist.
+    fn fetch_host_header(&self, height: Height) -> Option<MockHeader> {
```

Comment:
Leaving this comment here but it applies generally. We have a number of places with `fetch_<item>()`, `get_<item>()` or just `<item>()`. I personally prefer the last :)

```suggestion
    fn host_header(&self, height: Height) -> Option<MockHeader> {
```

Reply:
Agree with the fact we should be consistent in naming this class of functions. The simple name `<item>` seems like the best option to me also. Will rename this and keep an eye out for other spots.